src/gui/TrackContainerView.cpp

```diff
@@ -377,18 +378,46 @@ void TrackContainerView::dropEvent( QDropEvent * _de )
 		//it->toggledInstrumentTrackButton( true );
 		_de->accept();
 	}
-	else if( type == "samplefile" || type == "pluginpresetfile" 
-		|| type == "soundfontfile" || type == "vstpluginfile"
-		|| type == "patchfile" )
+	else if (type == "samplefile" || type == "pluginpresetfile"
+			|| type == "soundfontfile" || type == "vstpluginfile"
+			|| type == "patchfile")
 	{
-		InstrumentTrack * it = dynamic_cast<InstrumentTrack *>(
-				Track::create( Track::InstrumentTrack,
-								m_tc ) );
-		PluginFactory::PluginInfoAndKey piakn =
-			pluginFactory->pluginSupportingExtension(FileItem::extension(value));
-		Instrument * i = it->loadInstrument(piakn.info.name(), &piakn.key);
-		i->loadFile( value );
-		//it->toggledInstrumentTrackButton( true );
+		const TrackView * trackView = trackViewAt(_de->pos().y() - m_timeLineWidgetHeight);
+		//if we drop on a sample track, add sample TCO to it
+		if (type == "samplefile" && trackView && trackView->getTrack()->type() == Track::SampleTrack)
+		{
+			for (auto tv : trackViews())
+			{
+				if (tv == trackView)
+				{
+					int trackHeadWidth = ConfigManager::inst()->value("ui", "compacttrackbuttons").toInt()==1
+											 ? DEFAULT_SETTINGS_WIDGET_WIDTH_COMPACT + TRACK_OP_WIDTH_COMPACT
+											 : DEFAULT_SETTINGS_WIDGET_WIDTH + TRACK_OP_WIDTH;
+
+					int xPos = _de->pos().x() < trackHeadWidth
+											  ? trackHeadWidth
+											  : _de->pos().x();
+
+					SampleTCO * newTco = new SampleTCO(tv->getTrack());
+					newTco->setSampleFile(value);
+					newTco->movePosition(fixedTCOs() == false
+										 ? MidiTime((xPos - trackHeadWidth) / pixelsPerTact()
+													* MidiTime::ticksPerTact()).toNearestTact()
+										 : MidiTime(0)
+										   );
+				}
+			}
+		}
+		else
+		{
+			InstrumentTrack * it = dynamic_cast<InstrumentTrack *>(
```

Comment:
correct me if i'm wrong, isn't it another feature?

Reply:
This code handles the file types if we are not dropping on a sampletrack or it's not a sample file:
```
(type == "samplefile" || type == "pluginpresetfile"
			|| type == "soundfontfile" || type == "vstpluginfile"
			|| type == "patchfile")
```